### PR TITLE
[CID-1946] Calling service delete when receive project_destroy

### PR DESCRIPTION
**Pull request by ahmed-ali-55**
Jira: https://leanix.atlassian.net/browse/CID-1946

Call `DELETE /services` On receiving project webhook with event `project_destroy`.

### CODE REVIEW

**Comment by geoandri** [comment]
Do we handle this on regular full scans? When we perform a full scan, do we omit the archived repositories?

File: src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/RepositoryService.kt [hunk 1]
```diff
@@ -44,6 +45,7 @@ class RepositoryService(
 
     fun saveAll(repositories: List<Repository>, assignment: GitLabAssignment, eventType: EventType) {
         repositories
+            .filter { !it.isArchived() }
```
Comment (henriqamaral) [inline src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/RepositoryService.kt:48]: I think today on VSM we show the archived ones.

File: src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/WebhookConsumerService.kt [hunk 2]
```diff
@@ -42,17 +43,31 @@ class WebhookConsumerService(
 
     private fun processProject(payload: String) {
         val project = mapper.readValue<ProjectChange>(payload)
+        val repository = gitlabProvider.getRepositoryByPath(project.pathWithNamespace)
```
[the remark this answers is not in the document]
Comment (ahmed-ali-55) [reply src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/WebhookConsumerService.kt]: yeah. moved back

File: src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/WebhookConsumerService.kt [hunk 3]
```diff
@@ -18,7 +18,9 @@ import net.leanix.vsm.gitlab.broker.shared.exception.NamespaceNotMatchException
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Service
 
-val PROJECT_EVENTS = listOf("project_create", "project_update", "project_rename", "project_transfer")
+const val PROJECT_DESTROY_WEBHOOK_EVENT_NAME = "project_destroy"
+val PROJECT_EVENTS =
```
Comment (henriqamaral) [inline src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/WebhookConsumerService.kt:22]: I think you can migrate the constants to this class:
https://github.com/leanix/vsm-gitlab-broker/blob/747365891db7036a477455943746950c483883e7/src/main/kotlin/net/leanix/vsm/gitlab/broker/shared/Constants.kt

Comment (ahmed-ali-55) [reply src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/application/WebhookConsumerService.kt]: done

**Review by henriqamaral — APPROVED** [review]

**Pull request merged**